File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.Stack;

class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        Stack<Integer> st = new Stack<>();
        
        for(int i=0; i<ingredient.length; i++){
            if(st.size()<3){
                st.push(ingredient[i]);
                continue;
            }else if(ingredient[i]==1){
                if(st.peek()==3){
                    st.pop();
                    if(st.peek()==2){
                        st.pop();
                        if(st.peek()==1){
                            st.pop();
                            answer++;
                            continue;
                        }
                        st.push(2);
                    }
                    st.push(3);
                }
            }
            st.push(ingredient[i]);
        }
        
        return answer;
    }
}

File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    var answer = 0;
    let stack = [];
    for(var i=0; i<ingredient.length; i++){
        let ing = ingredient[i];
        stack.push(ing);
        if(ing === 1 && stack.slice(-4).join('') === '1231') {
            answer++;
            for(var j=0; j<4; j++) stack.pop();
        }
    }
    return answer;
}